File: MATLAB/example_specifyTrackingWeights.m
%% import necessary classes
import rraTools.*   

% setup some values that we may want to iterate through
config.participant = 'Hamner2010_v4_subject01';
config.conditions = {'Run_20002','Run_30002','Run_40002','Run_50002'}
config.trialnums = 1:3

c = 1 % set condition for this example (Run_20002)
t = 1 % set trial nubmer for this example
%% Instantiate the rrasetup object
        
mydir = pwd; % get current folder (where code is running from)
cd('..\') % move up one folder
% get system path to data folder
trialpath = fullfile(pwd, 'HamnerOpt\subject01\',config.conditions{c},['Trial_' num2str(t)]);
cd(mydir) % move back to code folder

r = rrasetup(trialpath,config.participant,[]);
% update file names from defaults
r.fileset.kinfile = [config.conditions{c}, '_IK.mot'];
r.fileset.grffile = [config.conditions{c}, '_GRF.mot'];
r.toolsettings.kinfile = [config.conditions{c}, '_IK.mot'];

%% write tasks with user specified tracking weights
myWeights = {'ankle_angle_r','knee_angle_r','hip_flexion_r','ankle_angle_l','knee_angle_l','hip_flexion_l';...
    25,10,5,10,5,15}; % all other weights will be set to 1
r.writeTasksFile('Kp',1600,'UserWeights',myWeights) 
% write external loads file
r.writeExtLoads()
% write ReservesFile to match specified model. Change the value
% after 'ResidualForce' to increase or decrease the optimal force
% of residual actuators (FX, FY, FZ, MX, MY, MZ). Change the value
% after 'ReserveForce' to increase or decrease the ideal torque
% actuators applied at each joint.
r.writeReservesFile('ReserveForce',2000,'ResidualForce',75)
%% Run the first iteration of RRA 
r = r.runInitialRRA()

File: Python/example_specifyTrackingWeights.py


# %% load the class
import reduceresiduals

# %% setup some variables
# update trial path to match where the test data is saved on your system       
trialpath = "C:/Users/Jordan/Documents/PhD/rra_tools/rra-optimization/HamnerOpt/subject01/Run_40002/Trial_1" # full path to trial folder. must contain scaled model, and ik data
participant = "Hamner2010_v4_subject01" # first part of the model name
condition = [] #empty for this test data, but is otherwise the second part of the model name constructed as "participant_condition.osim"

# %% initialize an instance of the class
rraopts = reduceresiduals.rrasetup( trialpath, participant, condition )

# %% set properties different from defaults
rraopts.fileset.kinfile = "Run_40002_IK.mot"
rraopts.fileset.grffile = "Run_40002_GRF.mot"
rraopts.toolsettings.kinfile = rraopts.fileset.kinfile

myWeights = {"ankle_angle_r": 15,"ankle_angle_l": 5,"knee_angle_r": 5,"knee_angle_l": 10,"hip_flexion_r": 25,"hip_flexion_l": 10}
rraopts.createTasksFile(UserWeights=myWeights) # all other weights will be set to 1
# %% first pass at RRA. Weights are already written so we don't need to set createTasks = True
rraopts.initialRRA(createReserves = True, createExtLoads = True)

# %% can finish running the mass iterations, and TWSA below if desired
